### PR TITLE
Issue #174: context menu could be modified by grid settings

### PR DESCRIPTION
**Pull request by OgreTransporter**
It's possible to setup your own context menu without editing the source of jstree-grid (see #174)

### CODE REVIEW

**Comment by deitch** [comment]
In principle, I like this. Actually, I like it a lot. Here are some changes I would do:

* Yes, `contextmenu` is taken, but `contextitems` doesn't clarify what it is. I think `gridcontextmenu` is clearer.
* Need to add description to README
* I don't think the default, if no `contextitems` (or `gridcontextmenu`) is defined should be the `edit` menu; I think it should be no special menu. Allowing a special function called `edit` (e.g. `gridcontextmenu: 'edit'`) is fine, but if nothing is defined, then no menu at all.

As long as the context menu for the grid was fixed, then making it dependent on whether or not the jstree had a `contextmenu` defined made sense (sort of). With proper definition and control, removing that default makes more sense.

So those three changes, and then we can merge in. 

Thoughts?

**Comment by OgreTransporter** [comment]
I would leave the default context menu to make it similar to jsTree. In jsTree you also have an option to switch it on/off and a second option for menu items. Now, my pull request has the same features. jsTree also has a default context menu.

I would not bind the grid context menu to the main tree context menu. It's not useful when I just want to edit the grid and not the main tree. In this case I could disable the tree context menu and only use grid context menu.

**Comment by deitch** [comment]
I like everything, and I agree wholeheartedly that grid context menu should not be bound to main tree context menu.

The only question I have is having a menu at all by default. In jstree, you actually need to enable the `"contextmenu"` plugin to get a context menu. Here, it is _always_ on by default. It also is impossible to override it, because if I set `gridcontextmenu` to `false` or `null`,  the logic of `s.gridcontextmenu || function (grid,tree,node,val,col,t,target) {...}` means it will always have that menu.

How about we do it as defaulting to nothing, no special `gridcontextmenu`. If the user defines a function, then we use that function. But, we offer the user a default edit one if they set `gridcontextmenu` to `true`?

Examples in the code inline...

File: jstreegrid.js [hunk 1]
```diff
@@ -170,7 +170,19 @@
 					sortAsc: true,
 					fixedHeader: s.fixedHeader || true,
 					width: s.width,
-					height: s.height
+					height: s.height,
+					gridcontextmenu : s.gridcontextmenu || function (grid,tree,node,val,col,t,target)
```
Comment (deitch) [inline jstreegrid.js]: Per the comment, if we save the whole function as a `defaultContextMenu`. Then:

````javascript
// in the above
gridcontextmenu : s.gridcontextmenu

// and after
if (gridcontextmenu === true) {
  gridcontextmenu = function(grid, tree,node,val,col,t,target) {...}
} else if (!!gridcontextmenu === false) {
  gridcontextmenu = false;
} 
````

File: jstreegrid.js [hunk 2]
```diff
@@ -838,12 +850,7 @@
 				return function (e) {
 					if (gs.context) {
```
Comment (deitch) [inline jstreegrid.js]: And here:

````javascript
if (gs.gridcontextmenu) {...
````

**Comment by OgreTransporter** [comment]
Ok, I've changed the Default property handling.

**Comment by deitch** [comment]
So looking at the code, it looks like the 3 scenarios are:

* If `gridcontextmenu` is `undefined` (or `null` or `false`), no menu (just like jstree)
* If `gridcontextmenu` is provided, then it uses that menu (just like jstree)
* If `gridcontextmenu` is explicitly the boolean `true`, then it provides the default edit menu?

**Comment by OgreTransporter** [comment]
Yes, that was the idea.

**Comment by deitch** [comment]
Makes a lot of sense. I like it! :-)

I have one minor comment, just on the read me.

File: README.md [hunk 3]
```diff
@@ -94,6 +94,7 @@ We use the div to control the entire height and width, and the span to get acces
 * `draggable`: true/false if the columns should be draggable (requires jQuery UI with sortable plugin). Defaults to false.
 * `stateful`: true/false. If true, then whenever a column width is resized, it will store it in html5 localStorage, if available. Defaults to false.
 * `contextmenu`: true/false whether or not a context menu for editing the cells should be shown on right-click. Defaults to false.
+* `gridcontextmenu`: function to create context menu items (see context menu of jstree).
```
Comment (deitch) [inline README.md]: Can we add here: 

````markdown
* `gridcontextmenu`: function to create context menu items; see context menu of jstree for format. In addition, if `false` or not set, no special context menu. If `true`, creates a default menu to edit the cell entry.
````

**Comment by deitch** [comment]
Added. Just a few more sentences to clarify the README, then let us merge in.

**Comment by OgreTransporter** [comment]
I've changed the readme

**Comment by deitch** [comment]
> I've changed the readme

And I've been slow in getting back to you. Client work has been crazy busy. All good, of course, but takes away from the fun of open source. 

Let's take a look right now.

**Comment by deitch** [comment]
And done. Thanks!! I need to update the version number and push to npm, doing now.

**Pull request merged**